src/formatDate.js

```diff
@@ -50,7 +50,25 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const result = [];
+  const dateObject = {};
+  const splitedData = date.split(fromFormat[3]);
+
+  for (let i = 0; i < splitedData.length; i++) {
+    dateObject[fromFormat[i]] = splitedData[i];
+  }
+
+  if ('YYYY' in dateObject) {
+    dateObject['YY'] = dateObject['YYYY'].slice(2);
+  }
+
+  dateObject['YYYY'] = +dateObject['YY'] < 30 ? `20${dateObject['YY']}` : `19${dateObject['YY']}`;
```

Comment:
The code uses the magic number '30'. Instead of hardcoding this value, consider creating a named constant to enhance code readability and maintainability. And in general, try to avoid using mystery numbers, please